pkg/test/expectations/expectations.go

```diff
@@ -129,9 +104,36 @@ func ExpectCleanedUp(c client.Client) {
 	}
 }
 
-func AttemptProvisioning(c client.Client, provisioner *v1alpha2.Provisioner, pod *v1.Pod) *v1.Pod {
-	ExpectCreatedWithStatus(c, pod)
-	ExpectCreated(c, provisioner)
-	ExpectEventuallyReconciled(c, provisioner)
-	return ExpectPodExists(c, pod.GetName(), pod.GetNamespace())
+func ExpectProvisioningSucceeded(c client.Client, controller controllers.Controller, provisioner *v1alpha2.Provisioner, pods ...*v1.Pod) []*v1.Pod {
+	for _, pod := range pods {
+		ExpectCreatedWithStatus(c, pod)
+	}
+	ExpectReconcileSucceeded(controller, provisioner)
+	result := []*v1.Pod{}
+	for _, pod := range pods {
+		result = append(result, ExpectPodExists(c, pod.GetName(), pod.GetNamespace()))
```

Comment:
Is it necessary to check if `len(result) == len(pods)`?  In other words, can reconciliation succeed, but result in not all pods being bin-packed?